File: 0463-island-perimeter/0463-island-perimeter.js
/**
 * @param {number[][]} grid
 * @return {number}
 */
var islandPerimeter = function(grid) {
    let terra = 0, adjacentes = 0;

    for (let i = 0; i < grid.length; i++) {
        for (let j = 0; j < grid[0].length; j++) {
            if (grid[i][j] === 1) {
                terra++;

                if (i < grid.length - 1 && grid[i+1][j] === 1) adjacentes++;
                if (j < grid[0].length - 1 && grid[i][j+1] === 1) adjacentes++;
            }
        }
    }

    return 4 * terra - 2 * adjacentes;
};

File: 0463-island-perimeter/0463-island-perimeter.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        terra, adjacentes = 0, 0

        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == 1:
                    terra += 1

                    if i < len(grid) - 1 and grid[i+1][j] == 1:
                        adjacentes += 1
                    if j < len(grid[0]) - 1 and grid[i][j+1] == 1:
                        adjacentes += 1
        
        return 4 * terra - 2 * adjacentes